### PR TITLE
Fix interface conversion when Kind() == string but v.(string) panics

### PR DESCRIPTION
**Pull request by ghthor**
Added a type switch in the type cast to handle the case when the Kind == string but type casting to string will panic because it's a typed string value.

### CODE REVIEW

File: redactrus.go [hunk 1]
```diff
@@ -56,9 +57,20 @@ func (h *Hook) Fire(e *logrus.Entry) error {
 			}
 
 			// Redact based on value matching in Data fields
+			// Handle fmt.Stringer type
+			if vv, ok := v.(fmt.Stringer); ok {
+				e.Data[k] = re.ReplaceAllString(vv.String(), "$1[REDACTED]$2")
+				continue
+			}
```
Comment (whuang8) [inline redactrus.go:64]: Can you please add a test that covers this case?

File: redactrus_test.go [hunk 2]
```diff
@@ -151,3 +151,17 @@ func TestNilField(t *testing.T) {
...
+// Logrus fields can have re-typed strings so test we handle this edge case
+func TestTypedStringValue(t *testing.T) {
+	logEntry := &logrus.Entry{
+		Data: logrus.Fields{"TypedString": TypedString("kind is string")},
+	}
+	h = &Hook{RedactionList: []string{"foo"}}
+	err := h.Fire(logEntry)
+
+	assert.Nil(t, err)
+	assert.Equal(t, logrus.Fields{"TypedString": "kind is string"}, logEntry.Data)
```
Comment (whuang8) [inline redactrus_test.go]: Thoughts on testing the redaction functionality as well?

```suggestion
	h = &Hook{RedactionList: []string{"kind"}}
	err := h.Fire(logEntry)

	assert.Nil(t, err)
	assert.Equal(t, logrus.Fields{"TypedString": "[REDACTED] is string"}, logEntry.Data)
```

**Review by whuang8 — CHANGES_REQUESTED** [review]
Hey @ghthor! Really appreciate the PR :)

File: redactrus_test.go (hunk 2 above)
Comment (ghthor) [reply redactrus_test.go]: Yeah, that's a good idea.

File: redactrus.go (hunk 1 above)
Comment (ghthor) [reply redactrus.go:64]: Yep, was thinking the same

**Review by whuang8 — APPROVED** [review]
